### ISSUE TITLE
build field with `model_field.formfield`

### ISSUE THREAD

**Issue by tvboxme** [encode/django-rest-framework#6155]
## Checklist

- [x] I have verified that that issue exists against the `master` branch of Django REST framework.
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [ ] This is not a usage question. (Those should be directed to the [discussion group](https://groups.google.com/forum/#!forum/django-rest-framework) instead.)
- [x] This cannot be dealt with as a third party library. (We prefer new functionality to be [in the form of third party libraries](http://www.django-rest-framework.org/topics/third-party-resources/#about-third-party-packages) where possible.)
- [x] I have reduced the issue to the simplest possible case.
- [ ] I have included a failing test as a pull request. (If you are unable to do so we can still accept the issue.)

## Steps to reproduce
I wrote a customized model field that always act as a MultiChoiceField, and I defined it's `formfield` to return `forms.TypedMultipleChoiceField`.
Serializer always build my field as a ChoiceField, and I can not redefine it's `serializer_choice_field` because I have other field with choices in this serializer.
At last, I only can hard code a `serializer.MultiChoiceField` everywhere I use this customized field.
## Expected behavior
Serializer should discover if `field.formfield` is redefined and returns a multi choice field, and should not only depends on `serializer_choice_field` as a global setting.

## Actual behavior
as above.

**Comment by tvboxme** [comment]
@rpkilby 

1. Please notice that, line 1208 in this file, after find `choices` in `kwargs`, that `field_class` will be changed to `self.serializer_choice_field`.
2. Still there is no way to let `Serializer` recognize `MultipleChoiceField`.

**Comment by tvboxme** [comment]
@rpkilby 
I think `Serializer` should always know, a field with choices is defined as "Multiple". It should be smart.

**Comment by rpkilby** [comment]
> Please notice that, line 1208 in this file, after find `choices` in `kwargs`, that `field_class` will be changed to `self.serializer_choice_field`.

Thanks for the clarification, my initial response assumed you were overriding a relational field, not a standard field. 

> I think `Serializer` should always know, a field with choices is defined as "Multiple". It should be smart.

Standard model fields don't typically support multiple choices. e.g., a `models.IntegerField` is not capable of storing a list of multiple integers - the ORM would raise an exception upon saving. 

Can you be more specific about your use case? It's not clear what exactly you're trying to achieve. There is most likely a better solution to the issue/ 

Either way, I don't think the PR is something we would want to accept, as serializers map to models, not forms. My recommendation would be to manually declare the field on the serializer.

**Comment by rpkilby** [comment]
Your example is interesting, but relying on `formfield` in the implementation is definitely a nonstarter. 

The simplest path here is to manually declare the field on your serializers. Otherwise, I would  recommend overriding `build_standard_field` to handle your `BitChoiceField`. e.g., you could do something like the following (note that I haven't tested it, there are probably bugs):

```python
class MyModelSerializer(serializers.ModelSerializer):
    @contextmanager
    def patch_serializer_choice_field(self, field_class):
        original = self.serializer_choice_field
        try:
            self.serializer_choice_field = field_class
            yield
        finally:
            self.serialzier_choice_field = original

    def build_standard_field(self, field_name, model_field):
        if isinstance(model_field, BitChoiceField):
            with self.patch_serializer_choice_field(MultipleChoiceField):
                return super().build_standard_field(field_name, model_field)
        else:
            return super().build_standard_field(field_name, model_field)
```

**Comment by tvboxme** [comment]
Yes you are right, more relying on specified attribute names might be harmful. I will write a mixin for serializer I think.
